pkg/applicationserver/io/web/webhooks.go

```diff
@@ -332,6 +333,13 @@ func (w *webhooks) newRequest(ctx context.Context, msg *ttnpb.ApplicationUp, hoo
 	if !ok {
 		return nil, errFormatNotFound.WithAttributes("format", hook.Format)
 	}
+	if paths := hook.FieldMask.GetPaths(); len(paths) > 0 {
```

Comment:
This is what was requested in the issue, and it will work, but it will not be ideal. 

The `ApplicationUp` message has a `oneof` field called `up` - it contains the main contents of the message - the `JoinAccept`, the `UplinkMessage` and so forth. You can see a full list of viable field mask paths here:
https://github.com/TheThingsNetwork/lorawan-stack/blob/dbd03c4e5f3639dae5432b23c999a79937d37471/pkg/ttnpb/messages.pb.paths.fm.go#L752-L923

The problem is that by design `SetFields` won't allow two `oneof` fields to be provided. Specifically the following field mask is invalid:
- `up.join_accept.received_at`
- `up.uplink_message.decoded_payload`

Since `up` is a `oneof` field, it is impossible for it to be both `join_accept` and `uplink_message`, so providing both basically renders the webhook unusable (`SetFields` will always return an error).

We can avoid this issue by doing the following thing:
- You can retrieve all of the fields in the mask which are _not_ `up` using `ttnpb.ExcludeSubFields(paths, "up")`. These are to be kept in the field mask that we want to apply.
1. You then have to check which `oneof` field of `up` is set. This will be a switch akin to the one on L283.
2. Based on the active message, you can exclude paths which are not part of the active `oneof`. Consider building a function in `pkg/ttnpb/fieldmask_utils.go` for this.
3. Take the union between the fields from (1) and (2). Use that as a field mask.

Reply:
Regarding the testing:
According to
```
$ tools/bin/mage go:Cover && go tool cover -html coverage.out
```
there is no coverage of the file at least in go tests. I had trouble producing those from the ground up on my own, but I can do that with some guidance if necessary.

Regarding the improvement the description is clear, I'm on it.

Reply:
<img width="1544" alt="image" src="https://user-images.githubusercontent.com/36161392/183392000-c012470b-f7f3-4d63-8684-ce9a168b3ed5.png">
Make sure that `go:cover` actually did finish successfully. You can find the tests in `webhooks_test.go`.